lib/recursive-methods.rb

```diff
@@ -1,49 +1,134 @@
 # Authoring recursive algorithms. Add comments including time and space complexity for each method.
 
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity:  O(n) because it calls the function n times.
+# Space complexity: O(n^2)because it is taking the memory of each value and the memory for each call in the stack
 def factorial(n)
```

Comment:
The space complexity is O(n) because each method call goes on the stack, but you're not making a new array or hash or something with each method call.